src/test/WormholeJoin.t.sol

```diff
@@ -0,0 +1,569 @@
+        require((z = x + y) >= x);
+    }
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x);
+    }
+    function mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x);
+    }
+
+    function transfer(address dst, uint256 wad) external returns (bool) {
+        return transferFrom(msg.sender, dst, wad);
+    }
+    function transferFrom(address src, address dst, uint256 wad)
+        public returns (bool)
+    {
+        require(balanceOf[src] >= wad, "Dai/insufficient-balance");
+        if (src != msg.sender && allowance[src][msg.sender] != type(uint256).max) {
+            require(allowance[src][msg.sender] >= wad, "Dai/insufficient-allowance");
+            allowance[src][msg.sender] = sub(allowance[src][msg.sender], wad);
+        }
+        balanceOf[src] = sub(balanceOf[src], wad);
+        balanceOf[dst] = add(balanceOf[dst], wad);
+        return true;
+    }
+    function mint(address usr, uint256 wad) external  {
+        balanceOf[usr] = add(balanceOf[usr], wad);
+    }
+    function burn(address usr, uint256 wad) external {
+        require(balanceOf[usr] >= wad, "Dai/insufficient-balance");
+        if (usr != msg.sender && allowance[usr][msg.sender] != type(uint256).max) {
+            require(allowance[usr][msg.sender] >= wad, "Dai/insufficient-allowance");
+            allowance[usr][msg.sender] = sub(allowance[usr][msg.sender], wad);
+        }
+        balanceOf[usr] = sub(balanceOf[usr], wad);
+    }
+    function approve(address usr, uint256 wad) external returns (bool) {
+        allowance[msg.sender][usr] = wad;
+        return true;
+    }
+}
+
+contract DaiJoinMock {
+    VatMock public vat;
+    DaiMock public dai;
+
+    constructor(address vat_, address dai_) {
+        vat = VatMock(vat_);
+        dai = DaiMock(dai_);
+    }
+    uint256 internal constant RAY = 10 ** 27;
+    function mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x);
+    }
+    function join(address usr, uint256 wad) external {
+        vat.move(address(this), usr, mul(RAY, wad));
+        dai.burn(msg.sender, wad);
+    }
+    function exit(address usr, uint256 wad) external {
+        vat.move(msg.sender, address(this), mul(RAY, wad));
+        dai.mint(usr, wad);
+    }
+}
+
+contract WormholeJoinTest is DSTest {
+
+    Hevm internal hevm = Hevm(HEVM_ADDRESS);
+    WormholeJoin internal join;
+    VatMock internal vat;
+    DaiMock internal dai;
+    DaiJoinMock internal daiJoin;
+    address internal vow = address(111);
+
+    uint256 internal constant RAD = 10**45;
+
+    function setUp() public {
+        vat = new VatMock();
+        dai = new DaiMock();
+        daiJoin = new DaiJoinMock(address(vat), address(dai));
+        join = new WormholeJoin(address(vat), address(daiJoin), "L2DAI", "ETHEREUM");
+        join.file("line", "L2", 1_000_000 ether);
```

Comment:
Similar to the comment above, should we use specific lowercase domain names everywhere? Like in https://github.com/makerdao/dss-wormhole/blob/17483ccf355d2771c0087e6355915ef64c912f4e/src/test/WormholeOracleAuth.t.sol#L80

Reply:
Done as well, just prefer to not mention any specific L2. I've renamed it to "l2network"

Reply:
Cool, I updated the domain name used in WormholeOracleAuth.t to `l2Network` in https://github.com/makerdao/dss-wormhole/pull/22.